### PR TITLE
Add undocumented configuration parameter and explain in porting guide

### PR DESCRIPTION
**Pull request by skylerbunny**
##### SUMMARY

Changing the ansible_* facts to the ansible_facts.* namespace is described as a 'major change' in the 2.5 CHANGELOG entries, but its impact in the porting guide, nor the new configuration variable introduced by it in ansible.cfg are not actually described.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/porting_guide_2.5.rst

##### ANSIBLE VERSION
```
ansible 2.5.0b1 (detached HEAD a48cf0cb0d) last updated 2018/02/12 09:10:51 (GMT -700)
  config file = /Users/rsteinfeldt/Documents/discovery-ansible/ansible.cfg
  configured module search path = [u'/Users/rsteinfeldt/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/rsteinfeldt/Documents/ansible/src/ansible/lib/ansible
  executable location = /Users/rsteinfeldt/Documents/ansible/bin/ansible
  python version = 2.7.14 (default, Jan  6 2018, 12:16:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### CODE REVIEW

**Comment by bcoca** [comment]
It was not added to the porting guide, cause there is nothing to do.

It is documented, just not added to the example ansible.cfg
http://docs.ansible.com/ansible/devel/config.html#inject-facts-as-vars

**Comment by skylerbunny** [comment]
In the sense that there is no immediate change needed, I suppose this is true, but that's also true of with_* loops versus the new 'loop' type; the old is essentially a deprecation. If however someone does (as I have a few) 'when' conditionals based on Ansible facts in playbooks, doesn't this change mean that it would be a good idea to start using the 'new' variable locations?

I guess the reason I went that direction was the assumption that as a major change (and saying that preventing the facts from being created in the old location will someday happen/be the default), that this is essentially a deprecation, suggesting that a change in playbook writing would be advised when referencing ansible facts. If I'm wrong, no problem!

**Comment by bcoca** [comment]
Yes, its a good idea to start using `ansible_facts.` prefix, which is why i kept this open. Just know that this makes your playbooks non backward compatible. The old way is not deprecated (yet) but that is the plan in the future.

We had actually deprecated `with_` but decided to give it a reprieve at the last moment, the docs were already in at that point.

**Comment by skylerbunny** [comment]
Got it. We definitely want the .cfg addition then at the least (possible wording notwithstanding).

If there is a better place for this discussion than the porting guide, ( http://docs.ansible.com/ansible/latest/playbooks_variables.html#information-discovered-from-systems-facts perhaps?) let me know and I can see what I can do about explaining 'old and new'. If it's in that section, it probably would be as a note indicating a mapping between a couple examples, and that the 'new way' is encouraged if you are exclusively using 2.5+.

**Comment by bcoca** [comment]
it can stay in porting guide, i was mostly explaining why i had not added it in the first place

**Review by dharmabumstead — APPROVED** [review]
Doc updates look good to me. Still needs tech signoff from the core team

**Review by bcoca — APPROVED** [review]
just needs rebase as the guide page was moved

**Comment by acozine** [comment]
rebase_merge

**Comment by acozine** [comment]
Thanks @skylerbunny for adding this information to the documentation.

**Comment by gundalow** [comment]
Shippable hadn't been triggered so I closed and reopened.
Made minor tweak for formatting

**Comment by gundalow** [comment]
@acozine Will need backport through to and including 2.5

**Pull request merged**
